### PR TITLE
Add dev guideline for Singleton service and java client 

### PR DESCRIPTION
**Pull request by linr211**
[skip CI]
Add dev guideline for Singleton service and java client so that the contributor can follow to: 
a) Setup the dev environment;
b) Run  Singleton service
c) submit code change
etc.

### CODE REVIEW

File: g11n-ws/docs/developer_guide_service.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+## Developer's Guide for Singleton Service
+
+### How to setup Intellij IDEA for development
+
+You can flow below steps to quickly setup your Dev environment:
+
+1. Install JDK 1.8(or later version) and Gradle 4.10.3(or later version)
```
Comment (nannany00) [inline g11n-ws/docs/developer_guide_service.md]: Please remove "(or later version)" for both JDK and Gradle.

Comment (linr211) [reply g11n-ws/docs/developer_guide_service.md]: do you mean it doesn't support later version?

Comment (nannany00) [reply g11n-ws/docs/developer_guide_service.md]: yes, the compile failed when use JDK 10 or Gradle 5

Comment (linr211) [reply g11n-ws/docs/developer_guide_service.md]: fixed

**Comment by linr211** [comment]
[skip CI]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -23,6 +26,21 @@ We have sub projects in each branch:
 |   g11n-angular-client   | Singleton Angular 7 client code                                                                                 |
 |   devops                | CI and automation testing code, it is not a place to contribute code unless you want to add CI or test scripts. |
 
+Changes should be made on your own forked branch. PR should be rebased on top of one of above branches without multiple branches mixed into the PR. If your PR do not merge cleanly, use commands listed below to get it up to date.
+
+```
+# origin is the origin Singleton repository
+# upstream is your forked repository under your github account
```
Comment (nannany00) [inline CONTRIBUTING.md]: Please swich here:
origin should be your forked Singleton
upstream is vmware Singleton

Comment (Xiaochao8) [inline CONTRIBUTING.md]: I suggest switch "upstream " to "origin" in the above 2 sentences.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -23,6 +26,21 @@ We have sub projects in each branch:
...
+git checkout master
+git fetch origin
+git merge origin/master
+git push upstream master
+```
+
+Accordingly, for each client please check out to the sub branch and push to remote sub branch.
+
 
 ## Contribute flow
```
Comment (Xiaochao8) [inline CONTRIBUTING.md]: Change "Contribute" to "Contribution".

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -93,5 +111,24 @@ You can open a PR after you get pass result of CI pipeline. CI pipeline will run
...
+
+For Singleton service feature development, you can use IntelliJ IDEA or Eclipse to do development work and import the codes as gradle project. The coding style follow [JAVA style](https://petroware.no/javastyle.html).
+
+To build the code, you can use gradle's build task. After the build task is finished, you can find the build under the project '/build/libs/' path. Please refer to [singleton service guideline](https://github.com/vmware/singleton/tree/master/g11n-ws/docs/developer_guide_service.md) for details.
+
+Unit test cases should be added to cover the new code.
+
+#### Singleton Client
+
+For each specific client, it depends on the programing lanunages to use the according development tool, please follow the individual guidline.
```
Comment (Xiaochao8) [inline CONTRIBUTING.md]: "according development tool" => "corresponding development tools"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -93,5 +111,24 @@ You can open a PR after you get pass result of CI pipeline. CI pipeline will run
...
+
+[Java client guideline](https://github.com/vmware/singleton/tree/master/g11n-ws/docs/developer_guide_javaclient.md)
+
+[AngularJS  guideline](https://github.com/vmware/singleton/tree/master/g11n-ws/docs/developer_guide_angularjs.md)
```
Comment (Xiaochao8) [inline CONTRIBUTING.md]: AngularJS  => Angular
developer_guide_angularjs.md => developer_guide_angular.md

File: g11n-ws/docs/developer_guide_javaclient.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+## Developer's Guide for Singleton Java Client
+
+### How to setup Intellij IDEA for development
+
+You can follow below steps to setup your development environment:
+
+1. Install JDK 1.8(or later version) and Gradle 4.10.3(or later version)
```
Comment (Xiaochao8) [inline g11n-ws/docs/developer_guide_javaclient.md]: Should support only JDK 1.8 and gradle 4.10.3(or compatible version)

File: g11n-ws/docs/developer_guide_javaclient.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+2. Install Intellij IDEA community version and configure Gradle home
+
+3. Import Java client codes as Gradle project
+
+4. Run build.gradle with 'clean build' to generate the build
+
+After building, you can find the build of Singleton's Java client from the building output folder.
```
Comment (Xiaochao8) [inline g11n-ws/docs/developer_guide_javaclient.md]: The path is under the project '/build/libs/'

File: CONTRIBUTING.md (hunk 5 above)
Comment (Xiaochao8) [reply CONTRIBUTING.md]: AngularJS is different from Angular. AngularJS is out of date.

File: g11n-ws/docs/developer_guide_javaclient.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+## Developer's Guide for Singleton Java Client
+
+### How to setup Intellij IDEA for development
+
+You can follow below steps to setup your development environment:
+
+1. Install JDK 1.8(or later version prior to JDK 10) and Gradle 4.10.3(or later version prior to Gradle 5)
```
Comment (Xiaochao8) [inline g11n-ws/docs/developer_guide_javaclient.md]: Currently support only JDk 1.8 and gradle 4.10.3(compatible version).

File: g11n-ws/docs/developer_guide_service.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+## Developer's Guide for Singleton Service
+
+### How to setup Intellij IDEA for development
+
+You can flow below steps to quickly setup your Dev environment:
+
+1. Install JDK 1.8(or later version prior to JDK 10) and Gradle 4.10.3(or later version prior to Gradle 5)
```
Comment (Xiaochao8) [inline g11n-ws/docs/developer_guide_service.md]: support only JDK1.8?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -95,3 +113,22 @@ You can open a PR after you get pass result of CI pipeline. CI pipeline will run
...
+
+For Singleton service feature development, you can use IntelliJ IDEA or Eclipse to do development work and import the codes as gradle project. The coding style follow [JAVA style](https://petroware.no/javastyle.html).
+
+To build the code, you can use gradle's build task. After the build task is finished, you can find the build under the project '/build/libs/' path. Please refer to [singleton service guideline](https://github.com/vmware/singleton/tree/master/g11n-ws/docs/developer_guide_service.md) for details.
+
+Unit test cases should be added to cover the new code.
+
+#### Singleton Client
+
+For each specific client, it depends on the programing lanunages to use the according development tool, please follow the according guidline.
```
Comment (Xiaochao8) [inline CONTRIBUTING.md]: according development tool => corresponding development tools

Comment (Xiaochao8) [reply CONTRIBUTING.md]: according guidline => corresponding guidline.

File: CONTRIBUTING.md (hunk 3 above)
Comment (linr211) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 10 above)
Comment (linr211) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (linr211) [reply CONTRIBUTING.md]: fixed

File: g11n-ws/docs/developer_guide_javaclient.md (hunk 8 above)
Comment (linr211) [reply g11n-ws/docs/developer_guide_javaclient.md]: fixed

File: g11n-ws/docs/developer_guide_service.md (hunk 9 above)
Comment (linr211) [reply g11n-ws/docs/developer_guide_service.md]: fixed

File: g11n-ws/docs/developer_guide_javaclient.md (hunk 7 above)
Comment (linr211) [reply g11n-ws/docs/developer_guide_javaclient.md]: fixed

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -23,6 +26,21 @@ We have sub projects in each branch:
 |   g11n-angular-client   | Singleton Angular 7 client code                                                                                 |
 |   devops                | CI and automation testing code, it is not a place to contribute code unless you want to add CI or test scripts. |
 
+Changes should be made on your own forked branch. PR should be rebased on top of one of above branches without multiple branches mixed into the PR. If your PR do not merge cleanly, use commands listed below to get it up to date.
+
+```
+# origin is the origin Singleton repository
```
[the remark this answers is not in the document]
Comment (linr211) [reply CONTRIBUTING.md]: fixed

**Pull request merged**